### DISCUSSION TITLE
[NSFW] manyvids.com: How to disable the MUTE state permanently?

### DISCUSSION THREAD

**Discussion by Dean-Corso** [uBlockOrigin/uAssets#21336] [Q&A]
### Prerequisites

- [X] This is **NOT** a [YouTube](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [Facebook](https://www.github.com/uBlockOrigin/uAssets/issues/3367) or [Twitch](https://www.github.com/uBlockOrigin/uAssets/issues/5184) report.
- [X] I read and understand the [policy about what is a valid filter issue](https://github.com/uBlockOrigin/uAssets/blob/master/README.md#uassets).
- [X] I verified that this issue is not a duplicate. (Search [here](https://github.com/uBlockOrigin/uAssets/issues) to find out.)
- [X] I forced an update of my filter lists. (Click the ["Purge all caches"](https://github.com/gorhill/uBlock/wiki/Dashboard:-Filter-lists#purge-all-caches) button while holding the 'Shift' key, then click the ["Update now"](https://github.com/gorhill/uBlock/wiki/Dashboard:-Filter-lists#update-now) button.)
- [X] I did not remove any of the [default filter lists](https://github.com/uBlockOrigin/uAssets/assets/66517106/6e8d024a-5869-47cf-b142-b2357eb6c7d0), or I have verified that the issue was not caused by removing any of the default lists.
- [X] I did not enable additional filter lists, or I have verified that the issue still occurs without enabling additional filter lists.
- [X] I do not have custom filters/rules, or I have verified that the issue still occurs without custom filters/rules.
- [X] I am not using uBlock Origin (uBO) along with other content blocker extensions.
- [X] I have verified that the web browser's built-in blocker or [DNS blocking](https://en.wikipedia.org/wiki/DNS_blocking) (standalone or through a VPN) is not causing the issue.
- [ ] I did not answer truthfully to **ALL** the above checkpoints.

### URL(s) where the issue occurs.

https://www.manyvids.com/Vids/

### Description

Hi guys, so it's not really an filter issue but I would like to ask here to get maybe some help with my problem. When you visit that website using the link above then you get a list of video preview thumbnails which getting started when you hover with the mouse over any of them. If you do then you will see a speaker icon which is crossed out (disabled state) and you need to click on it to enable the sound of it. Now, when you hover again over the same thumb or any other then you need again to click on the speaker icon to enable the sound and this is really annoying to do that every time. I would like to find any way to make the enabled / disabled state permanently but how is the question. I think I need to modify some JS on fly about getMute / setMute etc.

Problem2: Also have the same sound trouble when you open any video site (click on any preview thumb to open the video on new page). The video is here also playing in mute mode and I also need to click on speaker button to enable the sound every time. In this case the mute information is stored in a session or local cookie which holds a value string with an value 0 or 1 (mute / not mute). I tried to change that using the "set-local-storage-item" & "set-session-storage-item" commands to remove it or to change it but without success. The cookie gets updated every time when I load the website what is pretty bad. I can not $remove$ it and I even can not change the value of that object to 1 because its not working to use that object style and I only can post empty object etc.

So I tried this so far without success:
```
manyvids.com##+js(set-session-storage-item, rmpVolume, $remove$)
manyvids.com##+js(set-local-storage-item, rmpVolume, $remove$)

manyvids.com##+js(set-session-storage-item, rmpVolume, {"value":"1"})
manyvids.com##+js(set-local-storage-item, rmpVolume, {"value":"1"})
```

Maybe anyone could check this out and create some working rules to prevent the auto mute function all the time on that website. Thank you.

### Other extensions used

none

### Screenshot(s)

<details>
<summary>Screenshot(s)</summary>

</details>

### Configuration

<details>
<summary>Details</summary>

```yaml


```
</details>

**Comment by ghost** [comment]
www.manyvids.com##+js(set, HTMLVideoElement.prototype.muted, false)

**Comment by Dean-Corso** [comment]
Thank you for that working rule @gwarser. Only issue now is that the auto play video function is no more working on the single video itself and you need to press play manually now. Could you tell me why the session / local rules aren't working in that case? What kind of other possible possibilities would be doable with uBO XY rules to make those MUTE changes?

PS: Are there any more detailed tutorials about uBO rules and how to use them? I know the uBO resources URLs but frankly I still don't know how they all work and how using them correctly for specific situations. Looking for any kind of source with detailed explanations with examples (codes of HTML / JS etc). Otherwise any test webpage would be also great to test all rules to see how they work, similar like some ADS test pages. Would be good if you or anyone else could mention any sources I should read / checkout / test to get a better understanding of all or most rules. Thank you.

**Reply by krystian3w** [reply]
Maybe enough is add to permanent allow autoplay with sound on that page (Firefox, maybe too Vivalidi, Brave, Opera).

Also check settings in AutoplayStopper MV2 to add prermanent exception (mostly Chrome if other chromium forks added this to own settings).

**Comment by Dean-Corso** [comment]
Thanks for that info @krystian3w. I see that auto play option in Firefox and did add MV to my list and now it seems to work to get no more un-muted on every single video page. Only in Chrome I don't get it work. Just see a sound option which is already enabled but when I load any MV video page in new tab then the sound is disabled (muted) but seems only to happen when using incognito mode in Chrome & private window in Firefox.

By the way, about the video preview thumbnails, so earlier the sound was always enabled when you did hover with the mouse over them and then they changed that. The reason seems to be that the playing sound of the video previews does not STOP always correctly and keeps playing in a loop without to end. Must be some coding bug in their code they didn't fixed and just disable the sound by default to hide the muted playing sound (video) in the background (hidden). After a while of browsing on any video profile and checking out some video previews I always notice that the CPU usage getting higher and higher and scrolling starts to lag just because of that hidden playing video/audio/media which don't stop and you need to refresh the website what is also bad because you get lost the page scrolling position when all new pages getting loaded by JS only.
